independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanInfo.java

```diff
@@ -314,6 +364,24 @@ boolean hasDefaultDestroy() {
         return bound;
     }
 
+    List<DecoratorInfo> getBoundDecorators() {
+        if (decoratedMethods.isEmpty()) {
+            return Collections.emptyList();
+        }
+        List<DecoratorInfo> bound = new ArrayList<>();
+        for (DecorationInfo decoration : decoratedMethods.values()) {
+            for (DecoratorInfo decorator : decoration.decorators) {
+                if (!bound.contains(decorator)) {
+                    bound.add(decorator);
+                }
+            }
+        }
+        // Sort by priority (highest goes first) and by bean class
+        Collections.sort(bound,
+                Comparator.comparing(DecoratorInfo::getPriority).reversed().thenComparing(DecoratorInfo::getBeanClass));
```

Comment:
I don't remember TBH. Maybe it's not needed anymore - it could be a remnant from one of the iteration. Let me check...

Reply:
Ah, it's actually necessary because when we generate the subclass constructor the delegate subclass of the first decorator (lowest priority value) needs a reference to the next decorator in the chain (higher priority), i.e. the decorators are instantiated in the reverse order. For example, the constructor of the subclass from the `MultipleDecoratorsTest` looks like:

```java
public MultipleDecoratorsTest_ToUpperCaseConverter_Subclass(CreationalContext var1, InjectableDecorator var2, InjectableDecorator var3) {
      CreationalContextImpl var4 = CreationalContextImpl.child(var1);
     // Create  AlphaConverterDecorator with priority 20
     DecoratorDelegateProvider.set(new AlphaConverterDecorator_10323069ad1f4550a36181dff7c11f77412cbd15_Delegate_Subclass(this));
      Object var5 = ((InjectableReferenceProvider)var2).get((CreationalContext)var4);
      DecoratorDelegateProvider.unset();
      this.2a678714ab75a1004d3fee390ffaf01415bb5b27 = var5;

      // Create BravoConverterDecorator with priority 2
      CreationalContextImpl var6 = CreationalContextImpl.child(var1);
      DecoratorDelegateProvider.set(new BravoConverterDecorator_10323069ad1f4550a36181dff7c11f77412cbd15_Delegate_Subclass((AlphaConverterDecorator)var5));
      Object var7 = ((InjectableReferenceProvider)var3).get((CreationalContext)var6);
      DecoratorDelegateProvider.unset();
      this.e9cc21df6ac2c3dbccd4662c7c4663e2c89b2a02 = var7;
      HashMap var8 = new HashMap(1);
      this.metadata = (Map)var8;
   }
```

We could sort the decorators just for this particular problem but it's IMO more consistent to use the same order accross the whole codebase.

Reply:
No that's fine, and makes perfect sense. (You might want to add a comment :-) ) Just noticed that elsewhere, they are sorted in opposite order (which also makes perfect sense there. I was just curious.)